### PR TITLE
Improve exception wrapping and disable wrapping by default.

### PR DESCRIPTION
**Pull request by isaacery**

### CODE REVIEW

File: src/enact/invocations.py [hunk 1]
```diff
@@ -65,6 +65,16 @@ def set_from(self: ResourceT, other: Any):
 class NativeException(ExceptionResource):
   """A python exception wrapped as a resource."""
 
+  def __init__(self, type_name: str, str_representation: str, traceback: str):
+    self._type_name = type_name
+    self._str_representation = str_representation
+    self._traceback = traceback
+    super().__init__(traceback)
```
Comment (lcrh) [inline src/enact/invocations.py:72]: You should probably pass this in as *args to the superclass, so it's compatible with normal exception functionality. I also don't think this is compatible with the resource interface, i.e., I think you won't be able to commit and checkout an exception of this type and keep all the detail.

File: tests/test_invocations.py [hunk 2]
```diff
@@ -118,6 +118,16 @@ def test_auto_input_kwargs(self):
...
       IntToStr()(1), '1')
 
+  def test_native_exception_str(self):
+    """Tests native exceptions string representation."""
+    type_name = 'ValueError'
+    str_representation = 'Bad thing happened.'
+    traceback = 'foo: 23'
+    exception = enact.NativeException(
+      type_name, str_representation, traceback)
+    self.assertEqual(
```
Comment (lcrh) [inline tests/test_invocations.py:131]: add commit / checkout

File: src/enact/invocations.py [hunk 3]
```diff
@@ -775,7 +781,8 @@ def invoke(
       except raise_on_errors:
         raise
       except Exception:  # pylint: disable=broad-exception-caught
-        if not builder.completed:
+        wrap_error = builder.exception_wrapped and not wrap_exceptions
+        if not builder.completed or wrap_error:
```
Comment (lcrh) [inline src/enact/invocations.py:785]: maybe rename `wrap_error` to `should_raise_wrapped_error`

File: tests/test_invocations.py (hunk 2 above)
Comment (isaacery) [reply tests/test_invocations.py:131]: Done.

File: src/enact/invocations.py (hunk 3 above)
Comment (isaacery) [reply src/enact/invocations.py:785]: Agreed, done.

File: src/enact/invocations.py (hunk 1 above)
Comment (isaacery) [reply src/enact/invocations.py:72]: Done.

**Review by lcrh — APPROVED** [review]

**Pull request merged**